packages/bot-web-ui/src/stores/main-content-store.js

```diff
@@ -30,6 +32,9 @@ export default class MainContentStore {
     @action.bound
     onMount() {
         window.addEventListener('resize', this.setContainerSize);
+        if (getSetting('active_tab')) {
+            this.active_tab = getSetting('active_tab');
+        }
```

Comment:
the `<MainContent>` component supports an `active_tab` prop, could we use that instead of updating the `active_tab` after mounting? Currently it'll first load the workspace tab, then switch to the chart tab if that was the most previous active tab. 

You can check line 174 in `packages/bot-web-ui/src/app.jsx`, you can determine which tab should be active and then pass it as a prop.

Reply:
Aaron I checked your solution but there is a problem. we need to load the workspace because the Blockly initially injects into the workspace and if we don't load the workspace we get some errors. but I pass the check cache function to the app.js and I think it works fine